test/include/seqan3/test/performance/naive_minimiser_hash.hpp

```diff
@@ -0,0 +1,143 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Mitra Darvish <mitra.darvish AT fu-berlin.de>
+ * \brief Provides seqan3::views::naive_minimiser_hash.
+ */
+
+#pragma once
+
+#include <range/v3/view/sliding.hpp>
+
+#include <seqan3/alphabet/concept.hpp>
+#include <seqan3/core/type_traits/range.hpp>
+#include <seqan3/range/views/detail.hpp>
+#include <seqan3/range/views/kmer_hash.hpp>
+#include <seqan3/range/views/minimiser.hpp>
+#include <seqan3/std/ranges>
+
+namespace seqan3::detail
+{
+// ============================================================================
+//  naive_minimiser_hash_fn (adaptor definition)
+// ============================================================================
+
+//!\brief views::naive_minimiser_hash's range adaptor object type (non-closure).
+struct naive_minimiser_hash_fn
+{
+    /*!\brief                Store the shape and the window size and return a range adaptor closure object.
+    * \param[in] shape       The seqan3::shape to use for hashing.
+    * \param[in] window_size The windows size to use.
+    * \throws std::invalid_argument if the size of the shape is greater than the `window_size`.
+    * \returns               A range of converted elements.
+    */
+    constexpr auto operator()(shape const & shape, uint32_t const window_size) const
+    {
+        return seqan3::detail::adaptor_from_functor{*this, shape, window_size};
+    }
+
+    /*!\brief            Call the view's constructor with the underlying view as argument.
+     * \param[in] urange The input range to process. Must model std::ranges::viewable_range and the reference type of the
+     *                   range must model seqan3::semialphabet.
+     * \param[in] k      The k-mer size to construct hashes for.
+     * \returns          A range of converted elements.
+     */
+    template <std::ranges::viewable_range urng_t>
+    //!\cond
+        requires semialphabet<reference_t<urng_t>>
+    //!\endcond
+    constexpr auto operator()(urng_t && urange, size_t const k) const noexcept
```

Comment:
Whats this operator for? Isn't it just hashing?

Reply:
Ups, leftover from copy'n'pasting of the naive kmer_hash :sweat_smile: